typescript/sdk/src/contracts.ts

```diff
@@ -53,30 +53,42 @@ function getFactory(
   return factories[key];
 }
 
-export function filterAddresses(
-  addresses: HyperlaneAddresses<any>,
+export function filterAddressesMap(
+  addressesMap: HyperlaneAddressesMap<any>,
   factories: HyperlaneFactories,
-): HyperlaneAddresses<any> {
-  return pick(addresses, Object.keys(factories));
+): HyperlaneAddressesMap<typeof factories> {
+  // Filter out addresses that we do not have factories for
+  const pickedAddressesMap = objMap(addressesMap, (_, addresses) =>
+    pick(addresses, Object.keys(factories)),
```

Comment:
Nit, hoist Object.keys(factories) to a var as it's used in several places, including in maps